src/jobs/jobs.controller.ts

```
@@ -362,20 +362,45 @@ export class JobsController {
         user.currentGroups.some((g) => this.accessGroups?.admin.includes(g))
       ) {
         // admin users
-        let jobUser: JWTUser | null = user;
-        if (!jobCreateDto.ownerUser) {
-          jobUser = null;
-        } else if (user.username != jobCreateDto.ownerUser) {
-          jobUser = await this.usersService.findByUsername2JWTUser(
-            jobCreateDto.ownerUser,
-          );
+        let jobUser: JWTUser | null = null;
+        if (jobCreateDto.ownerUser) {
+          if (user.username != jobCreateDto.ownerUser) {
+            jobUser = await this.usersService.findByUsername2JWTUser(
+              jobCreateDto.ownerUser,
+            );
+            if (jobUser === null) {
+              Logger.log(
+                "Owner user was not found, using current user instead.",
```

Should we throw an httpexception instead of just logging it?

---

Let's address this one later. See relevant issue https://github.com/SciCatProject/scicat-backend-next/issues/1776